### DISCUSSION TITLE
Timer for motion payload on

### DISCUSSION THREAD

**Discussion by nrtkin** [danube/node-red-contrib-smarthome-powerswitch#9] [Ideas]
Hi,
is it possible to add timer for motion payload on?
thank you

**Comment by danube** [comment]
> Hi, is it possible to add timer for motion payload on? thank you

Hi nrtkin! You mean to send a time value inside your message (to overrule the configured value), to control the timeout?

**Reply by nrtkin** [reply]
Yes, that’s correct.

**Reply by danube** [reply]
Nice idea, thanks for the input! Expect this feature with 1.0.6, which is already available on Github. Just struggling with npm... 🤯

**Reply by danube** [reply]
@nrtkin Now there's version 1.0.6 available in the Node-RED library. You can now add msg.timeout to your off message. As the timer starts on off message, msg.timeout can only be added to this off message. Hope this is what you expected. Please let me know your experience with the node. grz tom

**Reply by diputssi** [reply]
Brilliant!  I needed this last week and decided today I would put in a feature request.  Already here!  Yay.

**Reply by diputssi** [reply]
Just tried timeout feature using pipeline feature and it works great.  Couple things to note:

- Default behavior seems to be milliseconds even though the code seems to check config setting.  Doesn't seem to work.
- If set to zero, it ignores the value or sets it to infinity which means stay ON.

I use this in a subflow and pass the timer value as a subflow config parameter.  In the subflow I have function that takes the config parameter.  If 0, it sets it to 1 (i.e. 1 ms) and if it is anything else converts it to a valid number.  Here's the code inside the function:

// Create new timer message
var newMsg = {};

// Grab timeout value from subflow config property and convert to milliseconds
var timeout = env.get("OffTimer");
if (timeout != 0)
{
    newMsg.timeout=Math.abs(timeout*1000);    // convert config property to ms.
}
else
{
    // If none. set default 1 ms.
    // Basically turn off because no timer set.
    // 0 seems to always assume / infinity
    newMsg.timeout=1;
}

// Check current payload message
if (msg.payload == true)
{
    newMsg.topic = "motion";
    newMsg.payload = true
    return newMsg
}
else if (msg.payload == false)
{
    newMsg.topic = "motion";
    newMsg.payload = false
    return newMsg
}
else
{
    return Msg
}

**Reply by danube** [reply]
Thanks for sharing your code! Regarding your things to note, both points are mentioned in the doc:
`If the 'Off Payload' message contains such a parameter, the configured timeout will be overwritten with this value. It must be positive and is evaluated in miliseconds (1000 = 1s).`

I take your statement to be more clear by adding a statement what happens, if the value is invalid (which "zero" as well as any non-number is) and probably find a way to inform also on other positions.

Hope that satisfies. Thanks for participating, have fun! Tom
